conn.go

```diff
@@ -713,6 +713,9 @@ func (c *Conn) prepareStatement(ctx context.Context, stmt string, tracer Tracer)
 	if err != nil {
 		flight.err = err
 		flight.wg.Done()
+		if err.Error() == "context deadline exceeded" {
```

Comment:
instead of doing this we should compare the value of `context.ErrTimeout`, but I think in this case we can just always remove the value on errors